kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/data/DefaultJsonRow.java

```diff
@@ -140,17 +145,94 @@ private static Object decodeElement(JsonNode jsonValue, DataType dataType) {
             return jsonValue.booleanValue();
         }
 
+        if (dataType instanceof ByteType) {
+            throwIfTypeMismatch(
+                "byte",
+                jsonValue.canConvertToExactIntegral() &&
+                    jsonValue.canConvertToInt() && jsonValue.intValue() <= Byte.MAX_VALUE,
```

Comment:
need range check for negative values? Same below.

Reply:
Done + added tests